### PR TITLE
Updated example for queryStringJsonSchema

### PR DESCRIPTION
**Pull request by danechitoaie**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by RafaelGSS** [comment]
The current example is not wrong. Do you have any reasons to change it?

**Comment by jsumners** [comment]
I have to agree that the example is more complete even if the original is valid JSON Schema. However, @danechitoaie it would be good if you could explain why you feel this change is necessary and why did you not update the other simplified examples?

**Comment by danechitoaie** [comment]
My reasoning was that with this example you can figure out how to add the `required: []` part in case you want some query string params to be required. With current example initially I was under the impression that you can't and had to spend some time googling until I figured out that I can be done this way.

File: docs/Validation-and-Serialization.md [hunk 1]
```diff
@@ -163,8 +163,11 @@ const bodyJsonSchema = {
...
-  name: { type: 'string' },
-  excitement: { type: 'integer' }
+  type: 'object',
+  properties: {
+    name: { type: 'string' },
+    excitement: { type: 'integer' }
+  }
 }
 
 const paramsJsonSchema = {
```
Comment (jsumners) [inline docs/Validation-and-Serialization.md:173]: @danechitoaie would you mind updating this schema as well?

**Comment by jsumners** [comment]
> My reasoning was that with this example you can figure out how to add the `required: []` part in case you want some query string params to be required. With current example initially I was under the impression that you can't and had to spend some time googling until I figured out that I can be done this way.

That's fair. Unfortunately, by not providing any context for your PR you got caught up in a spate of "is this spam or not" type suspicion. A little explanation with your pull requests will help in the future.

**Comment by danechitoaie** [comment]
> That's fair. Unfortunately, by not providing any context for your PR you got caught up in a spate of "is this spam or not" type suspicion. A little explanation with your pull requests will help in the future.

Yea. Totally agree, just that I've opened this PR while writing code for a project I'm currently working on and being in a zoom call at the same time. Too many things at once. I'll try to make more time and provide more info from the start in the future.

File: docs/Validation-and-Serialization.md (hunk 1 above)
Comment (danechitoaie) [reply docs/Validation-and-Serialization.md:173]: done

**Review by mcollina — APPROVED** [review]
lgtm

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Review by jsumners — APPROVED** [review]
LGTM

File: docs/Validation-and-Serialization.md [hunk 2]
```diff
@@ -163,13 +163,19 @@ const bodyJsonSchema = {
 }
 
 const queryStringJsonSchema = {
-  name: { type: 'string' },
-  excitement: { type: 'integer' }
+  type: 'object',
```
Comment (Eomm) [inline docs/Validation-and-Serialization.md:166]: I would revert this change because this example was on purpose.
It is written before the code:

> or a simpler variation in which the `type` and `properties` attributes are forgone and the parameters are listed at the top level (see the example below).

Comment (danechitoaie) [reply docs/Validation-and-Serialization.md:166]: Then maybe have both examples? I can add the original back and also the current one?

**Comment by jsumners** [comment]
@Eomm I want to clean up this document completely. It should follow the same style as the Decorators document.

**Comment by Eomm** [comment]
Do you mean listing the datails of the method and then the examples?

**Comment by jsumners** [comment]
I mean a complete overhaul of the whole document.
